.github/workflows/ci-cd.yml

```
@@ -502,6 +508,12 @@ jobs:
       codecov-token: ${{ secrets.CODECOV_TOKEN }}
 
   check:  # This job does nothing and is only used for the branch protection
+
+    # Separate 'pull_request' check from other checks to avoid confusion in
+    # GitHub branch protection about which check is required when multiple
+    # events trigger this workflow.
+    name: >-
+      ${{ github.event_name == 'pull_request' && 'check' || 'check&nbsp;' }}
```

How about
```suggestion
    name: >-
      check${{
        github.event_name == 'push'
        && ' (optional)'
        || ''
      }}
```
to also handle other important events like `merge_group` and `schedule`?

---

```suggestion
    name: >-
      ${{ github.event_name == 'push' && 'check&nbsp;'' || 'check' }}
```